upring.js

```diff
@@ -257,8 +258,69 @@ UpRing.prototype.request = function (obj, callback, _count) {
 
 UpRing.prototype.requestp = promisify(UpRing.prototype.request)
 
-function retry (that, obj, callback, _count) {
-  that.request(obj, callback, _count)
+UpRing.prototype.fire = function (obj, callback, _count) {
+  callback = callback || this._fireCallback
+  if (!this.isReady) {
+    this.once('up', this.fire.bind(this, obj, callback))
+    return
+  }
+
+  if (this._hashring.allocatedToMe(obj.key)) {
+    this.logger.trace({ msg: obj }, 'local call')
+    this._dispatch(obj, dezalgo(callback))
```

Comment:
I think this should not call the callback. The callback should be called anyhow, as it has no feedback, but just error on delivery. In this case, the delivery is guaranteed.